src/ja/worker/config.py

```diff
@@ -10,27 +9,73 @@ class WorkerConfig(Config):
     Config for the worker client.
     """
 
+    def __init__(self, uid: str, ssh_config: SSHConfig, resource_allocation: ResourceAllocation):
+        """!
+        creates resource allocation instance
+        """
+        if uid is None:
+            raise ValueError("missing uid")
```

Comment:
The worker UID should be allowed to be None.
In that case the UID should be assigned automatically by the server.